drs_downloader/clients/terra.py

```diff
@@ -0,0 +1,163 @@
+            return None
+
+    async def sign_url(self, drs_object: DrsObject) -> DrsObject:
+        """No-op.  terra returns a signed url in `get_object` """
+
+        assert isinstance(drs_object, DrsObject), "A DrsObject should be passed"
+
+        data = {
+            "url": drs_object.id,
+            "fields": ["accessUrl"]
+        }
+        session = aiohttp.ClientSession(headers={
+            'authorization': 'Bearer ' + self.token,
+            'content-type': 'application/json'
+        })
+
+        async with session.post(url=self.endpoint, json=data) as response:
+            try:
+                self.statistics.set_max_files_open()
+                response.raise_for_status()
+                resp = await response.json(content_type=None)
+                assert 'accessUrl' in resp, resp
+                if resp['accessUrl'] is None:
+                    account_command = 'gcloud config get-value account'
+                    cmd = account_command.split(' ')
+                    account = subprocess.check_output(cmd).decode("ascii")
+                    raise Exception(
+                        f"A valid URL was not returned from the server.  Please check the access for {account}\n{resp}")
+                url_ = resp['accessUrl']['url']
+                drs_object.access_methods = [AccessMethod(access_url=url_, type='gs')]
+                return drs_object
+            except ClientResponseError as e:
+                drs_object.errors.append(str(e))
+                return drs_object
+            finally:
+                await session.close()
+
+    async def get_object(self, object_id: str) -> DrsObject:
+        """Sends a POST request for the signed URL, hash, and file size of a given DRS object.
+
+        Args:
+            object_id (str): DRS URI
+
+        Raises:
+            Exception: The request was rejected by the server
+
+        Returns:
+            DownloadURL: The downloadable bundle ready for async download
+        """
+        data = {
+            "url": object_id,
+            "fields": ["fileName", "size", "hashes"]
+        }
+        session = aiohttp.ClientSession(headers={
+            'authorization': 'Bearer ' + self.token,
+            'content-type': 'application/json'
+        })
+
+        async with session.post(url=self.endpoint, json=data) as response:
+            try:
+                self.statistics.set_max_files_open()
+                response.raise_for_status()
+                resp = await response.json(content_type=None)
+
+                md5_ = resp['hashes']['md5']
+                size_ = resp['size']
+                name_ = resp['fileName']
+                return DrsObject(
+                    self_uri=object_id,
+                    size=size_,
+                    checksums=[Checksum(checksum=md5_, type='md5')],
+                    id=object_id,
+                    name=name_,
+                    access_methods=[AccessMethod(access_url="", type='gs')]
```

Comment:
Is the following line still necessary here, given the "accessUrl" is no longer being retrieved at this point?
```
access_methods=[AccessMethod(access_url="", type='gs')]
```

Reply:
[Fixed](https://github.com/anvilproject/drs_downloader/blob/feature/manager/drs_downloader/clients/terra.py#L145-L151)